### PR TITLE
Add check for numeric ids to not convert strings into numbers

### PR DESCRIPTION
**Pull request by felixsch**
## Description

card: https://trello.com/c/ReYjExvX/1866-rmt-fix-management-of-custom-repositories
bug: https://bugzilla.suse.com/show_bug.cgi?id=1182736

This is a bug occuring if the first characters of an ID are numbers e.g `15ptf-repo`. In `ActiveRecord` the `find_by` searches for the numeric id and strings automatically translated to an integer and trailing characters get truncated. This results in odd behaviour.

* How to test the change:

```
1) bin/rmt-cli repos custom add "https://download.opensuse.org/repositories/server:/http/SLE_15/" 15sp1-basesystem-ptf
2) bin/rmt-cli repos custom list
3) bin/rmt-cli repos custom remove 15sp1-basesystem-ptf
4) bin/rmt-cli repos custom list
```

❤️ Thank you! ❤️

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> in the <code>package</code> directory to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/185932/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 52d3e14cd2aaa26c0ff7e9ad763c909738427454 into fec62a0d6c4adf73863ad1ad841abef1a8e6c4ae)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/185933/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fa254949fb97b6ee169f9b8aba11dbe7a2de0974 into fec62a0d6c4adf73863ad1ad841abef1a8e6c4ae)

File: lib/rmt/cli/repos_base.rb [hunk 1]
```diff
@@ -38,8 +38,18 @@ def change_repo(id, set_enabled, custom: false)
     puts set_enabled ? _('Repository by ID %{id} successfully enabled.') % { id: id } : _('Repository by ID %{id} successfully disabled.') % { id: id }
   end
 
+  def is_numeric_id?(str)
+    # Check if given string is a plain number without any additional characters
+    # like '15sp3-ptf-repo-id'.
+    !!Integer(str) rescue false
```
Comment (ngetahun) [inline lib/rmt/cli/repos_base.rb:44]: The `!!` are not doing anything here. Exceptions are not handled like a truthy/falsy value. `Integer(str) rescue false` would be doing the same exact thing.

**Review by ngetahun — CHANGES_REQUESTED** [review]

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/187263/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fa254949fb97b6ee169f9b8aba11dbe7a2de0974 into c9a2c0a25899c58917d216fb150094001d39fb0b)

File: lib/rmt/cli/repos_base.rb (hunk 1 above)
Comment (felixsch) [reply lib/rmt/cli/repos_base.rb:44]: It converts a `integer` into `bool` in the case where `Integer` does not throw. I added it to match the type of the rescue case. Do you think I should remove it?

Comment (ngetahun) [reply lib/rmt/cli/repos_base.rb:44]: Yeah, It confuses someone else who might have a look. `Integer` throws an exception if it can't be converted.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/187333/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fa254949fb97b6ee169f9b8aba11dbe7a2de0974 into 05412b52a3cbaf2f3886fe77f84685d3f71aee40)

File: lib/rmt/cli/repos_base.rb (hunk 1 above)
Comment (felixsch) [reply lib/rmt/cli/repos_base.rb]: I removed it!

**Review by ngetahun — APPROVED** [review]
LGTM!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/187334/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e01f680ea9c9686fde07e8c32f16cdb16baa9e57 into 05412b52a3cbaf2f3886fe77f84685d3f71aee40)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/187335/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fa200ffd2877ad1841a003929237da305e7d5e7a into 9502b3b8b05e8c8cdfc60a0407605bde36495a36)

**Pull request merged**